phpBB/phpbb/attachment/delete.php

```diff
@@ -0,0 +1,323 @@
+
+/**
+ * Attachment delete class
+ */
+
+class delete
+{
+	/** @var config */
+	protected $config;
+
+	/** @var driver_interface */
+	protected $db;
+
+	/** @var filesystem  */
+	protected $filesystem;
+
+	/** @var resync */
+	protected $resync;
+
+	/** @var string phpBB root path */
+	protected $phpbb_root_path;
+
+	/** @var array Attachement IDs */
+	protected $ids;
+
+	/** @var string SQL ID string */
+	private $sql_id;
+
+	/** @var string SQL where string */
+	private $sql_where = '';
+
+	/** @var int Number of deleted items */
+	private $num_deleted;
+
+	/** @var array Post IDs */
+	private $post_ids = array();
+
+	/** @var array Message IDs */
+	private $message_ids = array();
+
+	/** @var array Topic IDs */
+	private $topic_ids = array();
+
+	/** @var array Info of physical file */
+	private $physical = array();
+
+	/**
+	 * Attachment delete class constructor
+	 *
+	 * @param config $config
+	 * @param driver_interface $db
+	 * @param filesystem $filesystem
+	 * @param resync $resync
+	 * @param string $phpbb_root_path
+	 */
+	public function __construct(config $config, driver_interface $db, filesystem $filesystem, resync $resync, $phpbb_root_path)
+	{
+		$this->config = $config;
+		$this->db = $db;
+		$this->filesystem = $filesystem;
+		$this->resync = $resync;
+		$this->phpbb_root_path = $phpbb_root_path;
+	}
+
+	/**
+	 * Delete Attachments
+	 *
+	 * @param string $mode can be: post|message|topic|attach|user
+	 * @param mixed $ids can be: post_ids, message_ids, topic_ids, attach_ids, user_ids
+	 * @param bool $resync set this to false if you are deleting posts or topics
+	 *
+	 * @return int|bool Number of deleted attachments or false if something
+	 *			went wrong during attachment deletion
+	 */
+	public function delete($mode, $ids, $resync = true)
+	{
+		if (!$this->set_attachment_ids($ids))
+		{
+			return false;
+		}
+
+		$this->set_sql_constraints($mode);
+
+		// Collect post and topic ids for later use if we need to touch remaining entries (if resync is enabled)
+		$this->collect_attachment_info($resync);
+
+		// Delete attachments from database
+		$this->delete_attachments_from_db();
+
+		if (!$this->num_deleted)
+		{
+			return 0;
+		}
+
+		// Delete attachments from filesystem
+		$this->remove_from_filesystem();
+
+		// If we do not resync, we do not need to adjust any message, post, topic or user entries
+		if (!$resync)
+		{
+			return $this->num_deleted;
+		}
+
+		// No more use for the original ids
+		unset($ids);
+
+		// Update post indicators for posts now no longer having attachments
+		$this->resync->resync('post', $this->post_ids);
+
+		// Update message table if messages are affected
+		$this->resync->resync('message', $this->message_ids);
+
+		// Now update the topics. This is a bit trickier, because there could be posts still having attachments within the topic
+		$this->resync->resync('topic', $this->topic_ids);
+
+		return $this->num_deleted;
+	}
+
+	/**
+	 * Set attachment IDs
+	 *
+	 * @param array $ids
```

Comment:
If it can be anything else then an array, it would be nice if you could add a proper docblock here.